src/app/api/common/delegates/getDelegates.ts

```diff
@@ -121,16 +121,53 @@ async function getDelegates({
   page = 1,
   sort = "weighted_random",
   seed,
+  filters,
 }: {
   page: number;
   sort: string;
   seed?: number;
+  filters?: {
+    issues?: string;
+    stakeholders?: string;
+  };
 }) {
   const pageSize = 20;
   const { namespace, ui, slug } = Tenant.current();
 
   const allowList = ui.delegates?.allowed || [];
 
+  const topIssuesParam = filters?.issues || "";
+  const topIssuesQuery =
+    topIssuesParam && topIssuesParam !== ""
+      ? `
+      AND jsonb_array_length(s.payload -> 'topIssues') > 0
+      AND EXISTS (
+        SELECT 1
+        FROM jsonb_array_elements(s.payload -> 'topIssues') elem
+        WHERE elem ->> 'type' = '${topIssuesParam}'
+        AND elem ->> 'value' IS NOT NULL
+        AND elem ->> 'value' <> ''
+      )
+      AND s.dao_slug = '${slug}'
+    `
+      : "";
+
+  // TODO 1/2: There is an inconsistency between top stakeholders and top issues. Top issues are filtered by a value
+  // TODO 2/2 : where the top stakeholders are filtered on type. We need to make this consistent and clean up the data and UI.
+  const topStakeholdersParam = filters?.stakeholders || "";
+  const topStakeholdersQuery =
+    topStakeholdersParam && topStakeholdersParam !== ""
+      ? `
+      AND jsonb_array_length(s.payload -> 'topStakeholders') > 0
+      AND EXISTS (
+        SELECT 1
+        FROM jsonb_array_elements(s.payload -> 'topStakeholders') elem
+        WHERE elem ->> 'type' = '${topStakeholdersParam}'
+      )
+      AND s.dao_slug = '${slug}'
```

Comment:
Does this actually work if `topStakeholdersParam` is assigned an empty string?  I would have assumed a type would be null, as opposed to an empty string.